src/ngx_http_lua_semaphore.h

```diff
@@ -0,0 +1,49 @@
+
+/*
+ * Copyright (C) cuiweixie
+ */
+
+
+#ifndef _NGX_HTTP_LUA_SEMAPHORE_H_INCLUDED_
+#define _NGX_HTTP_LUA_SEMAPHORE_H_INCLUDED_
+
+
+#include "ngx_http_lua_common.h"
+
+#define ngx_http_lua_sem_dis_api_err(c) \
```

Comment:
Okay, I've checked the code base. Yes, we already have the macro `ngx_http_lua_check_context` and I suggest you make use of it instead of reinventing your own :)